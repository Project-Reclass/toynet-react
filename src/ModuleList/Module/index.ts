import Module from './Module';
export * from './Module';
export default Module;