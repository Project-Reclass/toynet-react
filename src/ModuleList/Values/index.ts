import Values from './Values';
export default Values;