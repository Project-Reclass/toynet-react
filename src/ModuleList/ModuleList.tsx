import React from 'react';
import Module, { IModule } from './Module/Module';

interface Data extends IModule {
  subModules: IModule[];
}

const ModuleList = ({ moduleData }: {moduleData: Data[]}) => {
  return (
    <div>
      {moduleData.map(data => (
        <Module {...data} />
      ))}
    </div>
  );
};

export default ModuleList;