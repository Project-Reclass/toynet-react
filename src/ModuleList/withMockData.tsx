/* eslint-disable no-magic-numbers */
import React from 'react';
import { ModuleTypes, IModule } from './Module/Module';

const mockData = [
  {
    id: 1,
    moduleId: 0,
    title: 'Sandbox Tutorial',
    progress: 80,
    type: ModuleTypes.PARENT,
    subModules: [
      {
        id: 1,
        moduleId: 1,
        title: 'Emulator',
        progress: 40,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 1,
        title: 'Emulator Practice',
        progress: 10,
        type: ModuleTypes.EMULATOR,
      },
      {
        id: 1,
        moduleId: 1,
        title: 'Quiz',
        progress: 100,
        type: ModuleTypes.QUIZ,
      },
    ],
  },
  {
    id: 2,
    moduleId: 0,
    title: 'Switches and Routers',
    progress: 0,
    type: ModuleTypes.PARENT,
    subModules: [
      {
        id: 1,
        moduleId: 2,
        title: 'Switch',
        progress: 0,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Quiz',
        progress: 0,
        type: ModuleTypes.QUIZ,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Routers',
        progress: 0,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Quiz',
        progress: 0,
        type: ModuleTypes.QUIZ,
      },
    ],
  },
];

interface Data extends IModule {
  subModules: IModule[];
}

export default (Component: React.ComponentType<{moduleData: Data[]}>) => () => {
  return <Component moduleData={mockData} />;
};