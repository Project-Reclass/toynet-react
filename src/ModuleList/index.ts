import withMockData from './withMockData';
import ModuleList from './ModuleList';

export default withMockData(ModuleList);