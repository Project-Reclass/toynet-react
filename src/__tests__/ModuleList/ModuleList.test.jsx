import React from 'react';

import renderer from 'react-test-renderer';
import ModuleList from 'src/ModuleList';

describe('The Module List', () => {
  it('should render and match snapshot', () => {
    const tree = renderer.create(<ModuleList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});