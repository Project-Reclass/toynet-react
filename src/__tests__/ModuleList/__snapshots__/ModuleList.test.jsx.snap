// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Module List should render and match snapshot 1`] = `
<div>
  <div
    class="css-ojm4ju"
  >
    <div
      class="css-1djaw1e"
    >
      <div
        class="css-1w1vbxm"
      >
        <div
          class="css-qx149r"
        >
          <h2
            class="css-10ycv28"
          >
            Welcome!
          </h2>
          <p
            class="css-1q2wm2p"
          >
            <p
              class="css-fzcsno"
            >
              Welcome to the demo of our 
              <span
                class="css-1t24val"
              >
                Networking Fundamentals
              </span>
               course. You can click on each module's chevron to see its submodules. You must go through the modules and their submodules in order. To start or revisit a submodule, click 
               
              <a
                class="css-1im2cym"
              >
                Go To Submodule 
                 &gt;.
              </a>
            </p>
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-8z0fpj"
    >
      <div>
        <div
          class="css-11dqbsu"
        >
          <svg
            class="css-1axzmpk"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
          <div
            class="css-z7mtfw"
          >
            <p
              class="css-grke7g"
            >
              Example Module
            </p>
            <p
              class="css-kiw8ad"
            >
              2 / 3 completed, 2 in progress
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero css-vppuuz"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 250ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="css-0"
            >
              <p
                class="css-yy7hv6"
              >
                Computer need to move lots of information (e.g. emailed files, social media interactions, video streams) from one device to another. Whether itâ€™s through WiFi or over some kind of cable, computers communicate with a series of bits (0s and 1s) in tiny chunks at a time. The most common way of sending these tiny chunks is with a packet. This module is all about packets! For details about the submodule, click on its star.
              </p>
              <div
                class="css-e19uy1"
              >
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-6i3b32"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-lfbim6"
                      href="/module/0/quiz/4001"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Quiz: Life of a Network Packet
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-6i3b32"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-lfbim6"
                      href="/module/0/article/2001"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Article: Ever wondered how underwater cables are laid?
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-zlze97"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-b1im7z"
                      href="/value/5001"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Value: Integrity
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="css-11dqbsu"
        >
          <svg
            class="css-1axzmpk"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
          <div
            class="css-z7mtfw"
          >
            <p
              class="css-grke7g"
            >
              Second Module
            </p>
            <p
              class="css-kiw8ad"
            >
              0 / 3 completed, 0 in progress
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero css-vppuuz"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 250ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="css-0"
            >
              <p
                class="css-yy7hv6"
              />
              <div
                class="css-e19uy1"
              >
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-zlze97"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-lfbim6"
                      href="/module/0/quiz/4002"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Quiz: Quiz 2
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-zlze97"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-lfbim6"
                      href="/module/0/article/2002"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Article: Father of the internet, Vint Cerf, on creating the interplanetary internet
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <svg
                    class="css-zlze97"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="css-gmscv5"
                  >
                    <a
                      class="css-b1im7z"
                      href="/value/5002"
                    >
                      <p
                        class="css-fzcsno"
                      >
                        Value: Values - Respect
                      </p>
                      <a
                        class="css-u5zpo1"
                      >
                        Go to Submodule &gt;
                      </a>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
